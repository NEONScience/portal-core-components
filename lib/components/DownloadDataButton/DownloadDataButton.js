"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _styles = require("@material-ui/core/styles");
var _Button = _interopRequireDefault(require("@material-ui/core/Button"));
var _SaveAlt = _interopRequireDefault(require("@material-ui/icons/SaveAlt"));
var _DownloadDataContext = _interopRequireDefault(require("../DownloadDataContext/DownloadDataContext"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
const _excluded = ["label"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.includes(n)) continue; t[n] = r[n]; } return t; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const DownloadDataDialog = /*#__PURE__*/_react.default.lazy(() => Promise.resolve().then(() => _interopRequireWildcard(require('../DownloadDataDialog/DownloadDataDialog'))));
const useStyles = (0, _styles.makeStyles)(() => ({
  gtmCaptureButton: {
    '& span': {
      pointerEvents: 'none'
    }
  }
}));
const DownloadDataButton = props => {
  const {
      label
    } = props,
    other = _objectWithoutPropertiesLoose(props, _excluded);
  const classes = useStyles();
  const [{
    dialogOpen,
    productData
  }, dispatch] = _DownloadDataContext.default.useDownloadDataState();
  function handleOpenDialog() {
    dispatch({
      type: 'setDialogOpen',
      open: true
    });
  }
  const gtmProps = {};
  if ((productData || {}).productCode) {
    gtmProps.className = classes.gtmCaptureButton;
    gtmProps['data-gtm-product-code'] = productData.productCode;
  }
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
    color: "primary",
    variant: "contained"
    // eslint-disable-next-line react/jsx-no-bind
    ,
    onClick: handleOpenDialog,
    "data-selenium": "download-data-button",
    endIcon: /*#__PURE__*/_react.default.createElement(_SaveAlt.default, null)
  }, gtmProps, other), label), !dialogOpen ? null : /*#__PURE__*/_react.default.createElement(_react.Suspense, {
    fallback: null
  }, /*#__PURE__*/_react.default.createElement(DownloadDataDialog, null)));
};
DownloadDataButton.propTypes = {
  label: _propTypes.default.string
};
DownloadDataButton.defaultProps = {
  label: 'Download Data'
};
const WrappedDownloadDataButton = _Theme.default.getWrappedComponent(DownloadDataButton);
var _default = exports.default = WrappedDownloadDataButton;