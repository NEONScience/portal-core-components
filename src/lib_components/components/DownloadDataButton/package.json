{
  "private": true,
  "name": "download-data-button",
  "main": "./DownloadDataButton.jsx",
  "module": "./DownloadDataButton.js"
}
