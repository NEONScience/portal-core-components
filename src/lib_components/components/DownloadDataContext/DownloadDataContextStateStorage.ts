import StorageService from '../../service/StorageService';

const key = 'downloadDataContextState';

const StateStorage = {
  saveState: (state: object) => StorageService.setObject(key, state),
  readState: (): object | null => StorageService.getObject(key),
};

Object.freeze(StateStorage);
export default StateStorage;
