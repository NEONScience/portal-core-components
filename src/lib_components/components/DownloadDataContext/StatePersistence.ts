import StateService from '../../service/StateService';

const key = 'downloadContextState';

export const persistState = (state: object) => {
  StateService.setObject(key, state);
};

export const readState = () => StateService.getObject(key);
