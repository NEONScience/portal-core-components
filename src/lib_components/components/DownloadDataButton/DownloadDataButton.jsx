import React, { Suspense } from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import DownloadIcon from '@material-ui/icons/SaveAlt';

import DownloadDataContext from '../DownloadDataContext/DownloadDataContext';
import Theme from '../Theme/Theme';

const DownloadDataDialog = React.lazy(() => import('../DownloadDataDialog/DownloadDataDialog'));

const useStyles = makeStyles(() => ({
  gtmCaptureButton: {
    '& span': {
      pointerEvents: 'none',
    },
  },
}));

const DownloadDataButton = (props) => {
  const {
    label,
    ...other
  } = props;

  const classes = useStyles();

  const [{ dialogOpen, productData }, dispatch] = DownloadDataContext.useDownloadDataState();

  function handleOpenDialog() {
    dispatch({ type: 'setDialogOpen', open: true });
  }

  const gtmProps = {};
  if ((productData || {}).productCode) {
    gtmProps.className = classes.gtmCaptureButton;
    gtmProps['data-gtm-product-code'] = productData.productCode;
  }

  return (
    <>
      <Button
        color="primary"
        variant="contained"
        onClick={handleOpenDialog}
        data-selenium="download-data-button"
        endIcon={<DownloadIcon />}
        {...gtmProps}
        {...other}
      >
        {label}
      </Button>
      {!dialogOpen ? null : (
        <Suspense fallback={null}>
          <DownloadDataDialog />
        </Suspense>
      )}
    </>
  );
};

DownloadDataButton.propTypes = {
  label: PropTypes.string,
};

DownloadDataButton.defaultProps = {
  label: 'Download Data',
};

const WrappedDownloadDataButton = Theme.getWrappedComponent(DownloadDataButton);

export default WrappedDownloadDataButton;
