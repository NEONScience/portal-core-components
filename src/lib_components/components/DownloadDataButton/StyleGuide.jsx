/* eslint-disable react/jsx-one-expression-per-line, jsx-a11y/anchor-is-valid */
import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';

import DocBlock from '../../../components/DocBlock';
import CodeBlock from '../../../components/CodeBlock';
import ExampleBlock from '../../../components/ExampleBlock';

import DownloadDataButton from './DownloadDataButton';

import DownloadDataContext from '../DownloadDataContext/DownloadDataContext';
import Theme from '../Theme/Theme';

import sampleProductData from '../../../sampleData/DP1.10017.001.json';
import sampleProductDataAeronet from '../../../sampleData/DP1.00043.001.json';
import sampleProductDataAop from '../../../sampleData/DP1.30010.001.json';
import sampleProductDataAopOSPipeline from '../../../sampleData/DP1.30012.001.json';
import sampleProductDataBold from '../../../sampleData/DP1.20105.001.json';
import sampleProductDataMGRast from '../../../sampleData/DP1.10107.001.json';
import sampleProductDataNPN from '../../../sampleData/DP1.10055.001.json';
import sampleProductDataPhenocam from '../../../sampleData/DP1.00033.001.json';

const useStyles = makeStyles((theme) => ({
  divider: {
    margin: theme.spacing(3, 0),
  },
}));

export default function StyleGuide() {
  const classes = useStyles(Theme);
  const downloadDataContextLink = (
    <Link
      href="#DownloadDataContext"
    >
      Download Data Context
    </Link>
  );

  return (
    <>

      <DocBlock>
        A button to launch the Download Data Dialog.
      </DocBlock>
      <CodeBlock>
        {`
import DownloadDataButton from 'portal-core-components/lib/components/DownloadDataButton';
        `}
      </CodeBlock>

      <Typography variant="h5" component="h3" gutterBottom>Usage</Typography>

      <DocBlock>
        The Download Data Button depends on the {downloadDataContextLink} for all
        state. If the context has no props other than <tt>productData</tt> then
        the Download Data Button will appear clickable and the <b>Download Data Dialog</b> it
        opens when clicked will contain no selections.
      </DocBlock>
      <ExampleBlock>
        <DownloadDataContext.Provider productData={sampleProductData.data}>
          <DownloadDataButton />
        </DownloadDataContext.Provider>
      </ExampleBlock>
      <CodeBlock>
        {`
import DownloadDataButton from 'portal-core-components/lib/components/DownloadDataButton';
import DownloadDataContext from 'portal-core-components/lib/components/DownloadDataContext';

const productData = {
  productCode: 'DPX.00001.001',
  productName: '2D wind speed and direction',
  siteCodes: [
    {
      "siteCode": "ABBY",
      "availableMonths": [ "2016-04", "2016-05", ... ],
    },
    {
      "siteCode": "BART",
      "availableMonths": [ "2014-10", "2014-12", ... ],
    },
    ...
  ],
  ...
};

<DownloadDataContext.Provider productData={productData}>
  <DownloadDataButton />
</DownloadDataContext.Provider>
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Custom Label</Typography>

      <DocBlock>
        Pass the <tt>label</tt> prop to customize the button text.
      </DocBlock>
      <ExampleBlock>
        <DownloadDataContext.Provider productData={sampleProductData.data}>
          <DownloadDataButton label="Fetch My Data" />
        </DownloadDataContext.Provider>
      </ExampleBlock>
      <CodeBlock>
        {`
import DownloadDataButton from 'portal-core-components/lib/components/DownloadDataButton';
import DownloadDataContext from 'portal-core-components/lib/components/DownloadDataContext';

const productData = {...};

<DownloadDataContext.Provider productData={productData}>
  <DownloadDataButton label="Fetch My Data" />
</DownloadDataContext.Provider>
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Dynamic Download Step Generation</Typography>

      <DocBlock>
        The dialog launched by the download button has all necessary logic encapsulated within
        to determine the set of required download steps, if any, and present a meaningful
        download workflow for all types of data products. See examples below for variation
        in the download workflow.
      </DocBlock>
      <ExampleBlock>
        <div>
          <DownloadDataContext.Provider productData={sampleProductDataAop.data}>
            <DownloadDataButton label="Download AOP Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataAopOSPipeline.data}>
            <DownloadDataButton label="Download AOP Data (published via OS Pipeline)" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataAeronet.data}>
            <DownloadDataButton label="Download Aeronet Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductData.data}>
            <DownloadDataButton label="Download AmeriFlux Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataBold.data}>
            <DownloadDataButton label="Download Bold Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataMGRast.data}>
            <DownloadDataButton label="Download MGRast Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataNPN.data}>
            <DownloadDataButton label="Download NPN Data" />
          </DownloadDataContext.Provider>
          <br /><br />
          <DownloadDataContext.Provider productData={sampleProductDataPhenocam.data}>
            <DownloadDataButton label="Download Phenocam Data" />
          </DownloadDataContext.Provider>
        </div>
      </ExampleBlock>

    </>
  );
}
