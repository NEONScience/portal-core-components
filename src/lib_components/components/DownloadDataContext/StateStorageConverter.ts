// eslint-disable-next-line import/no-cycle
import DownloadDataContext from './DownloadDataContext';

/**
 * Alter the current state for valid JSON serialization. Set objects
 * must be converted to Array objects for serialization.
 * @param currentState The current state
 */
const convertStateForStorage = (state: any): any => {
  if (state.fromAOPManifest) {
    // AOP S3 Files will incur too much data to be saved in session state
    // Restore to default state in terms of s3Files and selection state.
    return {
      ...state,
      s3FileFetches: { ...DownloadDataContext.DEFAULT_STATE.s3FileFetches },
      s3FileFetchProgress: DownloadDataContext.DEFAULT_STATE.s3FileFetchProgress,
      s3Files: { ...DownloadDataContext.DEFAULT_STATE.s3Files },
      manifest: { ...DownloadDataContext.DEFAULT_STATE.manifest },
      allStepsComplete: DownloadDataContext.DEFAULT_STATE.allStepsComplete,
      sites: {
        ...state.sites,
        value: [...DownloadDataContext.DEFAULT_STATE.sites.value],
      },
    };
  }
  return state;
};

const convertAOPInitialState = (state: any, propsState: any): any => {
  if (!state.fromAOPManifest) return state;
  return {
    ...state,
    s3FileFetches: { ...propsState.s3FileFetches },
    s3FileFetchProgress: propsState.s3FileFetchProgress,
    s3Files: { ...propsState.s3Files },
    manifest: { ...propsState.manifest },
    allStepsComplete: propsState.allStepsComplete,
    policies: { ...propsState.policies },
  };
};

// eslint-disable-next-line import/prefer-default-export
export { convertStateForStorage, convertAOPInitialState };
