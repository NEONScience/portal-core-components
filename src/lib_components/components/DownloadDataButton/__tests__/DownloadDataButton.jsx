import React from 'react';
import renderer from 'react-test-renderer';
import DownloadDataContext from '../../DownloadDataContext/DownloadDataContext';

import DownloadDataButton from '../DownloadDataButton';

jest.mock('../../DownloadDataContext/DownloadDataContext', () => ({
  ...(jest.requireActual('../../DownloadDataContext/DownloadDataContext').default),
  useDownloadDataState: jest.fn(),
}));

const { useDownloadDataState, DEFAULT_STATE } = DownloadDataContext;

describe('DownloadDataButton', () => {
  beforeEach(() => {
    useDownloadDataState.mockReset();
    useDownloadDataState.mockReturnValue([{
      ...DEFAULT_STATE,
    }]);
  });
  test('Renders correctly with only a label', () => {
    const tree = renderer
      .create(<DownloadDataButton label="foo" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Preserves MUI button props', () => {
    const tree = renderer
      .create((
        <DownloadDataButton
          label="foo"
          size="large"
          color="primary"
          variant="outlined"
          data-selenium="download"
        />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders with the dialog open from context state', () => {
    useDownloadDataState.mockReset();
    useDownloadDataState.mockReturnValue([{
      ...DEFAULT_STATE,
      dialogOpen: true,
    }]);
    const tree = renderer
      .create(<DownloadDataButton label="foo" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Automatically includes data-gtm props from product in context state', () => {
    useDownloadDataState.mockReset();
    useDownloadDataState.mockReturnValue([{
      ...DEFAULT_STATE,
      productData: {
        productCode: 'DPX.XXXXX.XXX',
      },
    }]);
    const tree = renderer
      .create(<DownloadDataButton label="foo" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
