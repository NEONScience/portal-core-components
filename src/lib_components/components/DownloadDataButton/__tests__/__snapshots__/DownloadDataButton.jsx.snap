// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadDataButton Automatically includes data-gtm props from product in context state 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary makeStyles-gtmCaptureButton-20 css-pbr1h0-MuiButtonBase-root-MuiButton-root"
  data-gtm-product-code="DPX.XXXXX.XXX"
  data-selenium="download-data-button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  foo
  <span
    className="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-yaa4uj-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SaveAltIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7zm-6 .67 2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`DownloadDataButton Preserves MUI button props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-colorPrimary css-cn9fhd-MuiButtonBase-root-MuiButton-root"
  data-selenium="download"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  foo
  <span
    className="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeLarge css-1cnbqr6-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SaveAltIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7zm-6 .67 2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`DownloadDataButton Renders correctly with only a label 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-pbr1h0-MuiButtonBase-root-MuiButton-root"
  data-selenium="download-data-button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  foo
  <span
    className="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-yaa4uj-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SaveAltIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7zm-6 .67 2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
      />
    </svg>
  </span>
</button>
`;

exports[`DownloadDataButton Renders with the dialog open from context state 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-pbr1h0-MuiButtonBase-root-MuiButton-root"
  data-selenium="download-data-button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  foo
  <span
    className="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-yaa4uj-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SaveAltIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7zm-6 .67 2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
      />
    </svg>
  </span>
</button>
`;
